sdk/iot/pnp/src/az_iot_pnp_client.c

```diff
@@ -53,32 +54,42 @@ AZ_NODISCARD az_result az_iot_pnp_client_init(
 
 AZ_NODISCARD az_result az_iot_pnp_client_get_user_name(
     az_iot_pnp_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name)
+    char* mqtt_user_name,
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
 {
   AZ_PRECONDITION_NOT_NULL(client);
-  AZ_PRECONDITION_VALID_SPAN(mqtt_user_name, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_mqtt_user_name);
+  AZ_PRECONDITION_NOT_NULL(mqtt_user_name);
+  AZ_PRECONDITION(mqtt_user_name_size > 0);
 
-  az_span remainder = mqtt_user_name;
-  az_span user_name;
-  AZ_RETURN_IF_FAILED(
-      az_iot_hub_client_user_name_get(&client->_internal.iot_hub_client, remainder, &user_name));
+  size_t written;
 
-  int32_t written = az_span_size(user_name);
-  remainder = az_span_slice_to_end(remainder, written);
+  // First get hub user name since it is unknown how long it will be
+  AZ_RETURN_IF_FAILED(az_iot_hub_client_user_name_get(
+      &client->_internal.iot_hub_client, mqtt_user_name, mqtt_user_name_size, &written));
 
-  int32_t required_length
-      = az_span_size(pnp_model_id) + az_span_size(client->_internal.root_interface_name) + 2;
+  az_span mqtt_user_name_span
+      = az_span_init((uint8_t*)mqtt_user_name, (int32_t)mqtt_user_name_size);
 
-  AZ_RETURN_IF_NOT_ENOUGH_SIZE(remainder, required_length);
+  int32_t required_length = (int32_t)written + (int32_t)sizeof(pnp_client_param_separator)
```

Comment:
Nit: I think this is `required_size` because we're checking it against the size input param?

Reply:
This currently matches the rest of the SDK and I think it's okay if it stays since it eventually gets assigned to `out_mqtt_topic_length` at the end. Seems to match.